actions/sync-calendar/gcal/testing/regen_replay.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"io/ioutil"
+	"log"
+
+	"cloud.google.com/go/httpreplay"
+	"github.com/ghodss/yaml"
+	"github.com/spinnaker/rotation-scheduler/actions/sync-calendar/gcal"
+	"github.com/spinnaker/rotation-scheduler/schedule"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/calendar/v3"
+	"google.golang.org/api/option"
+)
+
+var (
+	schedulePath = flag.String("schedule", "", "Required. Path to schedule YAML file.")
+	jsonKey      = flag.String("jsonKey", "", "Required. The path to the JSON key with access to the Calendar API.")
+	outPath      = flag.String("out", "", "Required. File path to the recorded replay file.")
+
+	calendarID = flag.String("calendarID", "spinbot@spinnaker.io", "Optional. "+
+		"The calendar ID to update. Defaults to spinbot@google.com, because this account's calendar is not actually "+
+		"used for anything, and it costs us to create additional users.")
+)
+
+// Records a real interaction with the GCP Calendar API for use in unit tests.
+//
+// Sample invocation from root of repo:
+// go run actions/sync-calendar/gcal/testing/regen_replay.go \
+//   --jsonKey ./rotation-scheduler.json \
+//   --schedule ./actions/sync-calendar/gcal/testing/test_schedule.yaml \
+//   --out ./actions/sync-calendar/gcal/testing/schedule.replay
+func main() {
+	flag.Parse()
```

Comment:
Might be overkill but seems like we could re-use the `main.go` flag validators here?

Reply:
I don't know of a way to import functions declared in a `main` package. Maybe @brandonnelson3 may know if this is possible or know of a technique to share flag validation logic?

Reply:
Merged the two binaries with a `--record` option